src/drivers/gps/gps.cpp

```diff
@@ -511,6 +512,21 @@ void GPS::handleInjectDataTopic()
 		return;
 	}
 
+	gps_inject_data_s msg;
+
+	// If there has not been a valid RTCM message for a while, try to switch to a different RTCM link
+	if ((hrt_absolute_time() - _last_rtcm_injection_time) > 5000000) {
+		for (uint8_t i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
+			if (_orb_inject_data_sub.ChangeInstance(i)) {
+				if (_orb_inject_data_sub.copy(&msg)) {
+					if ((hrt_absolute_time() - msg.timestamp) < 5000000) {
+						break;
+					}
+				}
+			}
+		}
```

Comment:
If there's no data available, each cycle will run through the subscriptions. It's not terrible, but you can avoid this by setting `_last_rtcm_injection_time = hrt_absolute_time();` and changing to the first instance in that case.

Reply:
In that case when the current RTCM source has disappeared and the second hasn't appeared, it will only check for the new source every 5 seconds. Which might not be bad.